### DISCUSSION TITLE
Using removeLayers with groups

### DISCUSSION THREAD

**Discussion by scadergit** [openlayers/openlayers#15522] [General]
I just want to verify that I'm not wrong to state the Map's `removeLayer` only works at the top (default) group level, and if I have additional layer groups then I have to recurse through each layer group and use the collection's `remove` to find the matching layer?

Update: I found that tracking the parent collection is easier for our application, but now I also wonder if the map's `change:layerGroup` event is recursive? I can't seem to get that event to trigger at all, let alone for sub layer groups.

**Comment by ahocevar** [comment]
Yes. The map's `removeLayer()` method acts on the map's root layer group. If you want to remove a layer from a group inside that group, you have to use `group.removeLayer()` .
